src/Model/Table/QueuedTasksTable.php

```diff
@@ -323,15 +324,12 @@ public function reset() {
 	 * @return bool Success
 	 */
 	public function markJobFailed($id, $failureMessage = null) {
-		$db = $this->getDataSource();
-		$fields = [
-			'failed' => 'failed + 1',
-			'failure_message' => $db->value($failureMessage),
-		];
-		$conditions = [
-			'id' => $id
-		];
-		return $this->updateAll($fields, $conditions);
+		$data=$this->get($id);
+
+		$data->failed++;
```

Comment:
by default the function `$this->getDataSource();` has been replaced in CakePHP 3